libfwupdplugin/tests/quirks.d/tests.quirk

```diff
@@ -20,4 +20,6 @@ Flags = updatable,internal
 Name = A25Lxxx
 CfiDeviceCmdChipErase = 0xc7
 CfiDeviceCmdSectorErase = 0x20
+CfiDeviceBlockSize = 0x200
```

Comment:
Well, I am a bit confused, does a block size should be bigger than sector size?

Reply:
also, does a page == sector?

my understanding is that the flash is divided into several blocks that are divided into several pages (i.e. sector?).

Reply:
> does a block size should be bigger than sector size

Ahh, no, normally a block is much smaller. i.e. lots of blocks in each sector.

Reply:
oh, so when splitting into chunks, the code below is wrong:

```
        chunks = fu_chunk_array_new(NULL,
                                    len,
                                    start_addr,
                                    self->flash_block_size,
                                    self->flash_sector_size);
```
except if `block_size` is bigger than `sector_size`; by that, I mean the last argument should be smaller than the one before; or maybe that is not important.

Reply:
> normally a block is much smaller

I don't think this has to be true; I think the other way words as well, but it would be unusual indeed.

Reply:
For SPI devices, normally the block size is something like 256 bytes, and the sector size is much larger, e.g. 4Kb. You can only erase in whole sector amounts, but you can normally write in either whole blocks, or even one byte at a time.

Reply:
maybe s/fu_cfi_device_get_block_size/fu_cfi_device_get_page_size might be more obvious?

Reply:
I've renamed this.

Reply:
I do not have a strong opinion; as long it is consistent with the name of the commands :/

Reply:
Well, I am still confused when reading the datasheets.

Look at section [3 MEMORY ORGANIZATION](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwiSkOuT4Nb0AhVLXhoKHZLOB2AQFnoECAYQAQ&url=https%3A%2F%2Fwww.gigadevice.com%2Fdatasheet%2Fgd25d40c%2F&usg=AOvVaw0I5W8LUwzN7g-AqxCslZFk)

Memory is organized into 8 blocks of 16 sectors.

Reply:
And a sector is composed of pages.

Reply:
so I guess, we may need to have three getters/setters in the end.

Reply:
re-reading, I will need also the block size.

I need the values 0x1.0000 and 0x1000 which are the block-size and sector-size for the C84016.